components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,340 @@
+                                "Rolling back updated Authenticator information.",
+                                existingAuthenticatorConfig.getName(), IdentityTenantUtil.getTenantDomain(tenantId)));
+            }
+            IdentityDatabaseUtil.rollbackTransaction(dbConnection);
+            ErrorMessages error = ErrorMessages.ERROR_WHILE_UPDATING_AUTHENTICATOR;
+            throw new AuthenticatorMgtServerException(error.getMessage(), error.getMessage(), error.getCode(), e);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(dbConnection);
+        }
+    }
+
+    @Override
+    public LocalAuthenticatorConfig getUserDefinedLocalAuthenticator(String authenticatorConfigName, Integer tenantId)
+            throws AuthenticatorMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+        try {
+            LocalAuthenticatorConfig config = getUserDefinedLocalAuthenticatorByName(
+                    dbConnection, authenticatorConfigName, tenantId);
+
+            return config;
+        } finally {
+            IdentityDatabaseUtil.closeConnection(dbConnection);
+        }
+    }
+
+    @Override
+    public List<LocalAuthenticatorConfig> getAllUserDefinedLocalAuthenticator(Integer tenantId)
+            throws AuthenticatorMgtException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+             NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                Query.GET_ALL_USER_DEFINED_AUTHENTICATOR_SQL)) {
+
+            statement.setString(Column.DEFINED_BY, DefinedByType.USER.toString());
+            statement.setInt(Column.TENANT_ID, tenantId);
+
+            List<LocalAuthenticatorConfig> allUserDefinedLocalConfigs = new ArrayList<>();
+            try (ResultSet rs = statement.executeQuery()) {
+                while (rs.next()) {
+                    LocalAuthenticatorConfig config;
+                    if (AuthenticationType.VERIFICATION.toString().equals(rs.getString(Column.AUTHENTICATION_TYPE))) {
+                        config = new VerificationAuthenticatorConfig();
+                    } else {
+                        config = new LocalAuthenticatorConfig();
+                    }
+                    config.setName(rs.getString(Column.NAME));
+                    config.setDisplayName(rs.getString(Column.DISPLAY_NAME));
+                    config.setEnabled(rs.getString(Column.IS_ENABLED).equals(IS_TRUE_VALUE));
+                    config.setDefinedByType(DefinedByType.valueOf(rs.getString(Column.DEFINED_BY)));
+                    allUserDefinedLocalConfigs.add(config);
+                }
+            }
+
+            return allUserDefinedLocalConfigs;
+        } catch (SQLException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(String.format("Error while retrieving the user defined local Authenticator for " +
```

Comment:
Update the debug log for the function context. This seems to be retrieving all authenticators

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8